src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -125,6 +126,15 @@ internal static SafeSslHandle AllocateSslContext(SslProtocols protocols, SafeX50
                         throw CreateSslException(SR.net_allocate_ssl_context_failed);
                     }
 
+                    if (!sslAuthenticationOptions.IsServer)
+                    {
+                        // The IdnMapping converts unicode input into the IDNA punycode sequence.
+                        string punyCode = new IdnMapping().GetAscii(sslAuthenticationOptions.TargetHost);
```

Comment:
Can `new IdnMapping` be cached in a static field rather than allocated on every call?

Reply:
IdnMapping is not documented as thread safe.

Reply:
It's not documented as such, but we also own the implementation and know it to be (at least as far as GetAscii is concerned), and they ship together as part of netcoreapp.  Maybe we should just doc it as such? cc: @tarekgh, @krwq